camp_fin/management/commands/import_transactions.py

```diff
@@ -53,38 +83,83 @@ def add_arguments(self, parser):
         )
 
     def handle(self, *args, **options):
-        if options["transaction_type"] not in ("EXP", "CON"):
+        transaction_type = options["transaction_type"]
+
+        if transaction_type not in ("EXP", "CON"):
             raise ValueError("Transaction type must be one of: EXP, CON")
 
         year = options["year"]
 
+        self.stdout.write(f"Loading data from {transaction_type}_{year}.csv")
+
+        quarters = {int(q) for q in options["quarters"].split(",")}
+        quarter_string = ", ".join(f"Q{q}" for q in quarters)
+
         with open(options["file"]) as f:
-            if options["transaction_type"] == "CON":
-                self.import_contributions(f, year)
+            self.stdout.write(
+                f"Importing transactions from filing periods beginning in {quarter_string}"
+            )
+
+            if transaction_type == "CON":
+                self.import_contributions(f, quarters, year, options["batch_size"])
+
+            elif transaction_type == "EXP":
+                self.import_expenditures(f, quarters, year, options["batch_size"])
 
-            elif options["transaction_type"] == "EXP":
-                self.import_expenditures(f, year)
+            self.stdout.write(self.style.SUCCESS("Transactions imported!"))
+
+        self.stdout.write(
+            f"Totaling filings from periods beginning in {quarter_string}"
+        )
+        self.total_filings(quarters, year)
+        self.stdout.write(self.style.SUCCESS("Filings totaled!"))
 
-        self.total_filings(year)
         call_command("aggregate_data")
 
-    def import_contributions(self, f, year):
+    def _records_by_filing(self, records, filing_quarters):
+        """
+        Group records by filing, then filter for filings beginning in the specified
+        quarter/s. Note that, because transactions are organized by year, transactions
+        for one filing can appear across two files, if the reporting period begins in
+        one year and ends in the next. This approach will return filings beginning in
+        the specified quarter in *any* year, so that these split cases will be covered.
+        For example, consider a filing period starting in December 2023 and ending in
+        February 2024. Transactions would be split across the 2023 and 2024 files. To
+        get them all, you would run the Q4 import for both 2023 and 2024.
+        """
+        return filter(
+            lambda x: get_quarter(x[0][2]) in filing_quarters,
+            groupby(tqdm(records), key=filing_key),
```

Comment:
this tqdm is a bit confusing, as it is over all the records, not just the filtered records. 

would something like

```python
return groupby(tqdm(filter(records, lambda x: ...)), key=filing_key)
```

work?

Reply:
Yup! Done.